server/src/test/java/org/elasticsearch/search/query/ExistsIT.java

```diff
@@ -227,4 +228,41 @@ public void testFieldAliasWithNoDocValues() throws Exception {
         assertSearchResponse(response);
         assertHitCount(response, 2);
     }
+
+    public void testJsonFields() throws Exception {
+        XContentBuilder mapping = XContentFactory.jsonBuilder().startObject()
+            .startObject("type")
+                .startObject("properties")
+                    .startObject("headers")
+                        .field("type", "json")
+                    .endObject()
+                .endObject()
+           .endObject()
+        .endObject();
+        assertAcked(prepareCreate("test").addMapping("type", mapping));
+
+        IndexRequestBuilder indexRequest = client().prepareIndex("test", "type", "1")
+           .setSource(XContentFactory.jsonBuilder()
+               .startObject()
+                   .startObject("headers")
+                       .field("content-type", "application/json")
+                   .endObject()
```

Comment:
Not especially related to the current PR, but how is case normalization being handled?  The normal HTTP header here is `Content-type` with a capital C; is there a lowercase filter being applied anywhere, or is that being ignored for the moment?